src/inmanta/server/server.py

```diff
@@ -1440,6 +1440,7 @@ def delete_project(self, project_id):
 
         environments = yield Environment.get_list(project=project.id)
         for env in environments:
+            yield self.agentmanager.stop_agents(env)
```

Comment:
perhaps it would be better to yield on both statements, so they run in parallel.  Both can take a while.